### PR TITLE
Add tests upgrade to v2.2.0 

### PR DESCRIPTION
**Pull request by anajuliabit**
This PR adds integration tests to insure protocol integrity on upgrade from 2.1.0 to 2.2.0

## Breaking changes
-   **PR**: Allows DR creation without the need for activation by protocol admin by @anajuliabit in https://github.com/bosonprotocol/boson-protocol-contracts/pull/466
      - [x] DR can be activated on creation.
- **PR**: Allowlist metatx methods by @zajck in https://github.com/bosonprotocol/boson-protocol-contracts/pull/461
   - [x] Meta transaction should work with allowlisted function
   - [x] Meta transaction should fail when function name is not allowlisted
   - [x] functionPointer positions should change
- **PR**: Raise and Escalate Dispute Orchestration by @cliffhall in https://github.com/bosonprotocol/boson-protocol-contracts/pull/489:
   - [x] Test if facets address pointed to selectors from OrchestrationHandlerFacet were replaced for facets OrchestrationHandlerFacet1 and OrchestrationHandlerFacet2
- **PR**: New voucher name and symbol by @zajck in https://github.com/bosonprotocol/boson-protocol-contracts/pull/477
  - [x] Test if new vouchers are created with new name and symbol
- **PR**: Rename operator to assistant #556
  - [x] Operator has been renamed to assistant
 
   
## New methods
- **PR**: Allowlist metatx methods by @zajck in https://github.com/bosonprotocol/boson-protocol-contracts/pull/461
   - [x] setAllowlistedFunctions
   - [x] isFunctionAllowlisted(bytes32)
   - [x] isFunctionAllowlisted(string)
   - [x] Meta transaction should fail when function name is not allowlisted
- **PR**: Add new facet ProtocolInitializationFacet by @anajuliabit in https://github.com/bosonprotocol/boson-protocol-contracts/pull/475
   - [x] getVersion
- **PR**: Init v2_2_0 by @zajck in https://github.com/bosonprotocol/boson-protocol-contracts/pull/492
   - [x] Should call initV2_2_0
- **PR**: Raise and Escalate Dispute Orchestration by @cliffhall in https://github.com/bosonprotocol/boson-protocol-contracts/pull/489
   - [x] raiseAndEscalateDispute
 - **PR**: Protocol support preminted voucher by @zajck in https://github.com/bosonprotocol/boson-protocol-contracts/pull/490
     - [x] setMaxPremintedVouchers
     - [x] getMaxPremintedVouchers
     - [x] commitToPreMintedOffer
     - [x] reserveRange
     - [x] createPremintedOfferWithCondition
     - [x] createPremintedOfferAddToGroup
     - [x] createPremintedOfferAndTwinWithBundle
     - [x] createPremintedOfferWithConditionAndTwinAndBundle
     - [x] createSellerAndPremintedOffer
     - [x] createSellerAndPremintedOfferWithCondition
     - [x] createSellerAndPremintedOfferAndTwinWithBundle
     - [x] createSellerAndPremintedOfferWithConditionAndTwinAndBundle
     - [x] createOfferWithCondition
     - [x] createOfferAddToGroup
     - [x] createOfferAndTwinWithBundle
     - [x] createOfferWithConditionAndTwinAndBundle
     - [x] createSellerAndOffer
     - [x] createSellerAndOfferWithCondition
     - [x] createSellerAndOfferAndTwinWithBundle
     - [x] createSellerAndOfferWithConditionAndTwinAndBundle
 - **PR**: Allow vouchers to be preMinted to voucher contract #563 
   - [x] reserveRange to contract

## Bug fixes
- **PR**: Fix bug on createTwinInternal by @anajuliabit in https://github.com/bosonprotocol/boson-protocol-contracts/pull/469
     - [x] Should ignore twin id set by seller and use nextAccountId on twin creation
- **PR**:Fix bug on tokenURI method on BosonVoucher #557 
   - [x] Should return the correct tokenURI when token is pre-minted

## Notes
- [x] Besides the missing checks above, we still have to investigate why upgrade-facets script is logging `undefined: 0xa7ef6a7c` when removing interfaces for DR, and why is removing DR interface

### CODE REVIEW

**Comment by zajck** [comment]
> 2. Investigate why upgrade-facets script is logging `undefined: 0xa7ef6a7c` when removing interfaces for DR, and why is removing DR interface

`0xa7ef6a7c` is `activateDisputeResolver` sigHash, which was removed in 2.2.0, so it's ok that's reported. It says `undefined` since upgrade script is not aware of actual function names that were previously registered on diamond. All it knows is that something was registered which is not in interface anymore and therefore needs to be removed.

**Review by zajck — DISMISSED** [review]

**Comment by mischat** [comment]
i will have to look at this later on now, but i will try and get my head around it :)

File: package-lock.json [hunk 1]
```diff
@@ -2113,14 +2064,14 @@
...
-      "integrity": "sha512-xQ6eUZl+RDyb/FiZe1h+U7qr/f4p/SrTSQcTPH2bjur3C5DbuW/zFgCU/b1P/xcIaEqJep+9ju4xDRi3rmChdQ=="
+      "version": "4.8.2",
+      "resolved": "https://registry.npmjs.org/@openzeppelin/contracts/-/contracts-4.8.2.tgz",
+      "integrity": "sha512-kEUOgPQszC0fSYWpbh2kT94ltOJwj1qfT2DWo+zVttmGmf97JZ99LspePNaeeaLhCImaHVeBbjaQFZQn7+Zc5g=="
     },
     "node_modules/@openzeppelin/contracts-upgradeable": {
-      "version": "4.7.1",
-      "resolved": "https://registry.npmjs.org/@openzeppelin/contracts-upgradeable/-/contracts-upgradeable-4.7.1.tgz",
-      "integrity": "sha512-5EFiZld3DYFd8aTL8eeMnhnaWh1/oXLXFNuFMrgF3b1DNPshF3LCyO7VR6lc+gac2URJ0BlVcZoCfkk/3MoEfg=="
+      "version": "4.8.2",
```
Comment (zajck) [inline package-lock.json:2072]: @openzeppelin/contracts-upgradeable should remain @ 4.7.1 
They made some breaking changes.

**Review by zajck — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (anajuliabit) [reply package-lock.json:2072]: fixed

**Review by zajck — DISMISSED** [review]

**Review by anajuliabit — COMMENTED** [review]
I can't approve but LGTM

**Review by zajck — DISMISSED** [review]

**Review by zajck — APPROVED** [review]

**Review by zajck — APPROVED** [review]

File: contracts/protocol/facets/ExchangeHandlerFacet.sol [hunk 2]
```diff
@@ -888,11 +875,7 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
      *
      * @return bool - true if buyer is authorized to commit
      */
-    function authorizeCommit(
-        address _buyer,
-        Offer storage _offer,
-        uint256 exchangeId
```
Comment (mischat) [inline contracts/protocol/facets/ExchangeHandlerFacet.sol:894]: I guess this should start with an underscore, _exchangeId ... but i don't want to block on this ... Can we changed in the next version

**Review by mischat — APPROVED** [review]
Looks good, only a minor comment, which we should address in a subsequent version perhaps

File: contracts/protocol/facets/ExchangeHandlerFacet.sol (hunk 2 above)
Comment (mischat) [reply contracts/protocol/facets/ExchangeHandlerFacet.sol:894]: @anajuliabit

**Pull request merged**